src/Tools/dotnet-monitor/CollectionRules/ExecuteAction.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal sealed class ExecuteAction : IAction<ExecuteOptions>
+    {
+        public async Task<ActionResponse> Execute(ExecuteOptions options, CancellationToken token)
+        {
+            ActionResponse executeResponse = new();
+
+            await Task.Run(() =>
```

Comment:
Wrapping this with Task.Run and immediately awaiting it only tosses the work to a thread pool thread. You might as well just do all of the work in the current thread and await for the process exit.